features/frameworks/greentea-client/greentea-client/test_env.h

```diff
@@ -28,6 +28,9 @@
 #endif
 
 #include <stdio.h>
+extern "C" {
```

Comment:
As @pan- suggested, one header file and make it compatible for both languages. Thats the entry point for tests (``test_env.h``). C++ can use both provided API, C only C part of it.